solution/backend/regulations/views/mixins.py

```diff
@@ -23,7 +23,8 @@ def get_context_data(self, **kwargs):
 class IsAuthenticatedMixin:
     def dispatch(self, request, *args, **kwargs):
         if not request.user.is_authenticated:
-            redirect_string = request.path + '?' + request.GET.urlencode()
-            return redirect(reverse('login') + '?next=%s' % redirect_string)
+
+            redirect_string = '?' + request.GET.urlencode() if request.GET else ''
+            return redirect(reverse('login') + '?next=%s' % request.path + redirect_string)
```

Comment:
I was noticing that the login redirect was appending a single `?` to the end of the redirect url; this should take care of that by only creating a `redirect_string` if there are query params.